### ISSUE TITLE
Global custom key_bindings?

### ISSUE THREAD

**Issue by BarbzYHOOL** [fish-shell/fish-shell#5311]
I moved all my functions and conf to /etc/fish instead of my user

But now my keybindings do not work anymore. I used to put them inside fish_user_key_bindings.fish but now they don't work

How can I solve this?

**Comment by faho** [comment]
Are they inside a function called `fish_user_key_bindings` or is just the file called that?

Is that the one taking precedence? Does `functions -D fish_user_key_bindings` give you the path of that one?

Alternatively, wait for 3.0.

**Comment by BarbzYHOOL** [comment]
> Are they inside a function called fish_user_key_bindings or is just the file called that?

Both, that's the name of the file and inside that function.

`functions -D fish_user_key_bindings`
No, it gives me that lol : /etc/fish/conf.d/fundle_plugins.fish

I use fundle to install plugins

(And yeah, i want to switch to 3.0 but I need to be sure I have backed everything up before etc)

**Comment by faho** [comment]
>No, it gives me that lol : /etc/fish/conf.d/fundle_plugins.fish I don't understand why

The conf.d scripts are loaded before config.fish, and that one hard-defines a function (i.e. without autoloading), so fish doesn't try to autoload it anymore since it already has a definition.

If you want, you can file a bug against fundle.

As a workaround, `source /etc/fish/functions/fish_user_key_bindings.fish` explicitly in your config.fish.

**Comment by BarbzYHOOL** [comment]
> If you want, you can file a bug against fundle.

No, he doesn't update it much but the problem comes from me anyway

> The conf.d scripts are loaded before config.fish, and that one hard-defines a function (i.e. without autoloading), so fish doesn't try to autoload it anymore since it already has a definition.

Ok I identified that I had added a specific key binding in my custom prompt and that was loaded instead. Is there a way for me to add a key binding in my custom prompt, but still load key bindings in /etc/fish ?

**Comment by faho** [comment]
>Is there a way for me to add a key binding in my custom prompt

Why would you add a key binding in a prompt?

I can only repeat myself: Before fish 3.0, key bindings have to go into fish_user_key_bindings. All others will be erased once the bindings are loaded, which happens after config.fish.

In 3.0 (and current master), key bindings go wherever. You can still use fish_user_key_bindings if you want, but you don't have to.

I still wouldn't recommend adding bindings in a prompt.

**Comment by BarbzYHOOL** [comment]
>Why would you add a key binding in a prompt?

because of another bug, I asked here and there was no solution. `Replace Alt+w by something that works on multiline prompts (still a bit buggy)`


So in 3.0 key bindings can be in different files and "multi loaded"?

**Comment by BarbzYHOOL** [comment]
That was my bug: https://github.com/fish-shell/fish-shell/issues/5033

And I had put the file there for sharing it to other people but I rather put it in the readme instead I guess :P
